metricflow/protocols/sql_client.py

```diff
@@ -10,15 +10,19 @@
 from pandas import DataFrame
 
 
-class SupportedSqlEngine(Enum):
-    """Enumeration of DB engines currently supported by MetricFlow"""
+class SqlEngine(Enum):
+    """Enumeration of SQL engines, including ones that are not yet supported."""
```

Comment:
As context, I was strict on this originally because it was causing problems on the closed source side - it was very hard to figure out what was actually working and what was not - and I did not want to duplicate that here. This also made it clear which implementations were supportable on closed source and which were not, since we had things like MySQL clients that looked an awful lot like a SqlClient but weren't really (and weren't meant to be, it's just there are only so many query interfaces one might need).

The other issue here is we always have to switch through the unsupported ones if we want to use the enum as a complete set. I don't know if this is a practical concern, though, since we pretty much never operate on more than one of these at a time outside of tests.